src/coreclr/jit/gentree.cpp

```diff
@@ -16921,6 +16921,55 @@ ssize_t GenTreeIndir::Offset()
     }
 }
 
+/** Test whether the two given nodes are the same leaves.
+ *  Right now, only constant integers and local variables are supported
+ */
+bool GenTree::NodesAreEquivalentLeaves(GenTree* tree1, GenTree* tree2)
+{
+    if (tree1 == nullptr && tree2 == nullptr)
+    {
+        return true;
+    }
+
+    // both null, they are equivalent, otherwise if either is null not equivalent
+    if (tree1 == nullptr || tree2 == nullptr)
+    {
+        return false;
+    }
+
+    tree1 = tree1->gtSkipReloadOrCopy();
+    tree2 = tree2->gtSkipReloadOrCopy();
+
+    if (tree1->TypeGet() != tree2->TypeGet())
+    {
+        return false;
+    }
+
+    if (tree1->OperGet() != tree2->OperGet())
+    {
+        return false;
+    }
+
+    if (!tree1->OperIsLeaf() || !tree2->OperIsLeaf())
+    {
+        return false;
+    }
+
+    switch (tree1->OperGet())
+    {
+        case GT_CNS_INT:
```

Comment:
Why not `GT_CNS_DBL` as well?
* A bitwise check catches the majority cases, without needing to worry about NaN and +0/-0

What about `GT_CNS_LNG` for 32-bit?
* Maybe a comment explaining it isn't covered because it is decomposed to two parts, if that's the reason...

Reply:
This is a general-purpose method on `GenTree` now, so I think we would definitely like to handle `GT_CNS_LNG` (that said, not handling something here should not be an issue correctness-wise).

One more note: `GenTreeIntCon` has additional fields: `gtCompileTimeHandle` and `gtFieldSeq`. Should they be compared here too (or maybe introduce `GenTreeIntCon::Equals`)?

(Also nit: `gtIconVal` -> `IconValue()`)

Reply:
This is no longer needed.